Assets/GlitchEscape/Scripts/Misc/FloatingText.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+using TMPro;
+
+public class FloatingText : MonoBehaviour
+{
+    private TextMeshProUGUI floatText;
+    private GameObject floatPannel;
+    private Transform floatTextArea;
+
+    void Awake()
+    {
+        // floatText = this.GetComponents<TextMeshProUGUI>();
+        floatText = gameObject.GetComponent<TextMeshProUGUI>();
+        floatPannel = this.transform.parent.gameObject;
+        floatPannel.SetActive(false);
+        floatTextArea = null;
+    }
+
+    void Update()
+    {
+        if (floatTextArea != null)
+        {
+            floatPannel.transform.position = Camera.main.WorldToScreenPoint(floatTextArea.position);
+        }
+    }
+
+    public void EnableText(Transform targetTransform, string text)
+    {
+        floatText.text = text;
+        floatTextArea = targetTransform;
+        floatPannel.SetActive(true);
+    }
+
+    public void DisableText(Transform targetTransform)
+    {
+        if (floatTextArea == targetTransform)
+        {
+            floatTextArea = null;
+            floatPannel.SetActive(false);
+        }
+    }
```

Comment:
lastly, the canvas + floating text setup needs to be added to the player prefab (specifically NewPlayer, or whatever we're currently using for that).

Do that, and we can just drag + drop the HumanTank prefab into any scene and it'll just work.

Also, in the future you'll need to add a mechanic to track human tank free-ing, etc., do this via a HumanTankController.cs script attached to the root player object that has the PlayerController.cs script attached.